tests/test_esm_runscripts/test_filedicts.py

```diff
@@ -409,47 +463,28 @@ def test_ln_raises_exception_when_target_path_does_not_exist(simulation_file, fs
     with pytest.raises(FileNotFoundError):
         simulation_file.ln("computer", "work")
 
-
 # ========== end of ln() tests ==========
 
 
-def test_check_path_in_computer_is_abs(fs):
+def test_check_path_in_computer_is_abs(simulation_file, fs):
     """
     Tests that ``esm_parser.user_error`` is used when the ``path_in_computer``
     is not absolute
     """
-
-    dummy_config = """
-    general:
-        thisrun_dir: "/work/ollie/pgierz/some_exp/run_20010101-20010101"
-        exp_dir: "/work/ollie/pgierz/some_exp"
-        thisrun_work_dir: "/work/ollie/pgierz/some_exp/run_20010101-20010101/work"
-    echam:
-        files:
-            jan_surf:
-                type: input
-                name_in_computer: T63CORE2_jan_surf.nc
-                name_in_work: unit.24
-                path_in_computer: pool/ECHAM/T63
-        experiment_input_dir: /work/ollie/pgierz/some_exp/input/echam
-        thisrun_input_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/echam
-    """
-    config = yaml.safe_load(dummy_config)
+    simulation_file.path_in_computer = Path("foo/bar")
```

Comment:
I'm okay with having a dictionary defined somewhere else in the tests to be used as default, but I don't like that the object is instanced outside of the test. IMO the `SimulationFile` object should be instanced in the test, after we have modified the default `dict` manually on the test too. This method is closer to our real case scenario: 1) user defines dict, 2) ESM-Tools handles that dict. Right now you are doing: 1) test defines dict, 2) test instances dict, 3) test modifies that instance using the object methods, 4) test handles the operation of the test

Reply:
I see your point, but the **unit test** aims to test a **specific** function/method behaviour with different edge cases. Testing the behaviour of the object is also important (eg. functional test). I propose to have the following:
- have very specific unit tests for every single method to increase the coverage.
- have buggy YAML files with descriptive names. Eg. `config_no_absolute_paths.yaml` and write **functional tests** that test the higher level behaviour of the object when these YAML files are parsed.

What do you guys think @mandresm @pgierz